steps/reasons-for-appealing/evidence-upload/EvidenceUpload.js

```diff
@@ -8,124 +10,97 @@ const maxFileSize = config.get('features.evidenceUpload.maxFileSize');
 const Joi = require('joi');
 const paths = require('paths');
 const formidable = require('formidable');
-const pt = require('path');
-const fs = require('fs');
 const moment = require('moment');
+const stream = require('stream');
 const request = require('request');
-const { get } = require('lodash');
 const fileTypeWhitelist = require('steps/reasons-for-appealing/evidence-upload/fileTypeWhitelist');
+const content = require('./content.en.json');
 
 const maxFileSizeExceededError = 'MAX_FILESIZE_EXCEEDED_ERROR';
 const wrongFileTypeError = 'WRONG_FILE_TYPE_ERROR';
+const fileMissingError = 'FILE_MISSING_ERROR';
 
-class EvidenceUpload extends Question {
+/* eslint-disable consistent-return */
+class EvidenceUpload extends AddAnother {
   static get path() {
     return paths.reasonsForAppealing.evidenceUpload;
   }
 
-  static makeDir(path, dirCallback) {
-    const p = pt.join(__dirname, path);
-    fs.stat(p, (fsError, stats) => {
-      if (fsError || !stats.isDirectory()) {
-        return fs.mkdir(p, dirCallback);
-      }
-      return dirCallback();
-    });
-  }
-
   static handleUpload(req, res, next) {
-    const pathToUploadFolder = './../../../uploads';
     const logger = Logger.getLogger('EvidenceUpload.js');
 
-    if (req.method.toLowerCase() === 'post') {
-      return EvidenceUpload.makeDir(pathToUploadFolder, mkdirError => {
-        if (mkdirError) {
-          return next(mkdirError);
-        }
-        const multiplier = 1024;
-        const incoming = new formidable.IncomingForm({
-          uploadDir: pt.resolve(__dirname, pathToUploadFolder),
-          keepExtensions: true,
-          type: 'multipart',
-          maxFileSize: maxFileSize * multiplier * multiplier
-        });
+    const urlRegex = RegExp(`${paths.reasonsForAppealing.evidenceUpload}/item-[0-9]*$`);
+    if (req.method.toLowerCase() === 'post' && urlRegex.test(req.originalUrl)) {
+      const multiplier = 1024;
+      const incoming = new formidable.IncomingForm({
+        keepExtensions: true,
+        type: 'multipart'
+      });
 
-        incoming.once('error', er => {
-          logger.info('error while receiving the file from the client', er);
-        });
+      incoming.onPart = part => {
+        const emptyRequestApproxSizeInBytes = 200;
+        if (incoming.bytesExpected <= emptyRequestApproxSizeInBytes) {
+          req.body = {
+            'item.uploadEv': fileMissingError
+          };
+          return next();
+        }
+        if (!part.filename) {
+          // let formidable handle all non-file parts
+          incoming.handlePart(part);
+          return;
```

Comment:
Don't think this ever gets hit but should this be 
return next();
Sorry think this is my bug.

Reply:
Yeah I think so, will change it.

Reply:
👍 I think it might be it once we have the textarea thingie?